openedx/core/djangoapps/credentials/management/commands/notify_credentials.py

```diff
@@ -164,12 +172,46 @@ def handle(self, *args, **options):
             'auto' if options['auto'] else 'manual',
         )
 
-        course_keys = self.get_course_keys(options['courses'])
-        if not (course_keys or options['start_date'] or options['end_date'] or options['user_ids']):
-            raise CommandError('You must specify a filter (e.g. --courses= or --start-date or --user_ids)')
+        program_course_run_keys = self._get_course_run_keys_for_programs(options["program_uuids"])
+
+        courses = options["courses"]
+        if not courses:
+            courses = []
+        if program_course_run_keys:
+            courses.extend(program_course_run_keys)
+
+        course_keys = self.get_course_keys(courses)
```

Comment:
Is `courses` at this point a mix of _course_ identifiers and _course_run_ identifiers? And is that mix a problem? It looks like the `get_course_keys` function is just expecting _course_ identifier strings, but I’m not sure. Maybe `CourseKeys` knows what to do with both?

Reply:
This is a regular issue in the LMS. The LMS is almost always using CourseKeys or string versions of the CourseKey object. I tend to use the `course_run` name for my variables because almost all keys in the platform are for course runs. The edx-platform codebase does not know what a Course is in our other systems.

Reply:
I did a quick rename and refactor that I hope will make this a little more clear in the code.